packages/test/test-end-to-end-tests/src/test/SummarizerWithDataStoreRealization.spec.ts

```diff
@@ -0,0 +1,278 @@
+    runtime.IFluidHandleContext.resolveHandle(request);
+
+const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+    [dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+    [dataStoreFactory2.type, Promise.resolve(dataStoreFactory2)],
+    ],
+);
+const containerRuntimeFactoryWithDefaultDataStore =
+    getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+    dataStoreFactory1,
+    registryStoreEntries,
+    undefined,
+    [innerRequestHandler],
+    runtimeOptions,
+);
+
+async function createSummarizer(
+    provider: ITestObjectProvider,
+    container: IContainer,
+    summaryVersion?: string,
+): Promise<ISummarizer> {
+    return createSummarizerFromFactory(
+        provider,
+        container,
+        dataStoreFactory1,
+        summaryVersion,
+        containerRuntimeFactoryWithDefaultDataStore,
+        registryStoreEntries,
+    );
+}
+
+function createDataStoreRuntime(factory: typeof FluidDataStoreRuntime = FluidDataStoreRuntime) {
+    return mixinSummaryHandler(async (runtime: FluidDataStoreRuntime) => {
+        const obj: Provider<SearchContent> = (await DataObject.getDataObject(runtime)) as Provider<SearchContent>;
+        const searchObj = obj.SearchContent;
+        if (searchObj === undefined) {
+            return undefined;
+        }
+        // ODSP parser requires every search blob end with a line-feed character.
+        const searchContent = await searchObj.getSearchContent();
+        if (searchContent === undefined) {
+            return undefined;
+        }
+        const content = searchContent.endsWith("\n") ? searchContent : `${searchContent}\n`;
+        return {
+            // This is the path in snapshot that ODSP expects search blob (in plain text) to be for components
+            // that want to provide search content.
+            path: ["_search", "01"],
+            content,
+        };
+    }, factory);
+}
+
+/**
+ * Validates whether or not a GC Tree Summary Handle should be written to the summary.
+ */
+describeNoCompat("Summary with Search Blobs and DataStore realization during Summarization",
+ (getTestObjectProvider) => {
+    let provider: ITestObjectProvider;
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject1;
+
+    const createContainer = async (): Promise<IContainer> => {
+        return provider.createContainer(runtimeFactory);
+    };
+
+    async function waitForSummary(summarizer: ISummarizer): Promise<string> {
+        // Wait for all pending ops to be processed by all clients.
+        await provider.ensureSynchronized();
+        const summaryResult = await summarizeNow(summarizer);
+        return summaryResult.summaryVersion;
+    }
+
+    describe("Realize DataStore during Search and make sure pendingSummaries does not get corrupted", () => {
```

Comment:
nit: This describe block seems unncessary as there is only one test.

Reply:
But it still needs a text, doesn't it ?